### PR TITLE
Enable new networks for Swaps

### PR DESCRIPTION
**Pull request by dan437**
## Explanation
This PR enables Arbitrum and Optimism for Swaps. Most of the work for this was done on backend, frontend just needs a few small changes to enable it. I've also added support for multi layer fee calculations for Optimism, which I got from the metamask-extension.

Inspired by similar PR for Avalanche: https://github.com/MetaMask/metamask-mobile/pull/3766

### CODE REVIEW

File: app/util/networks/index.js [hunk 1]
```diff
@@ -271,3 +282,56 @@ export function blockTagParamIndex(payload) {
...
+  const OVMGasPriceOracle = getContractFactory('OVM_GasPriceOracle').attach(
+    predeploys.OVM_GasPriceOracle,
+  );
+  const abi = JSON.parse(
+    OVMGasPriceOracle.interface.format(utils.FormatTypes.json),
+  );
+  return eth.contract(abi).at(OVMGasPriceOracle.address);
+};
+
+export const fetchEstimatedL1Fee = async (eth, txMeta) => {
```
Comment (dan437) [inline app/util/networks/index.js:331]: This function and functions it calls should be pretty much the same with the metamask-extension's code: https://github.com/MetaMask/metamask-extension/blob/develop/ui/helpers/utils/optimism/fetchEstimatedL1Fee.js#L18

File: ios/Podfile.lock [hunk 2]
```diff
@@ -854,7 +854,7 @@ SPEC CHECKSUMS:
   RNInAppBrowser: e36d6935517101ccba0e875bac8ad7b0cb655364
   RNKeychain: 4f63aada75ebafd26f4bc2c670199461eab85d94
   RNOS: 6f2f9a70895bbbfbdad7196abd952e7b01d45027
-  RNReanimated: 0a4de6844119f4f1c3a46e6e7958ee7425ab4d0f
+  RNReanimated: 33c3c887be53972b5944017f65abb06f9bad4cf0
```
Comment (gantunesr) [inline ios/Podfile.lock:857]: I don't this is necessary, you can revert it

File: app/util/networks/index.js [hunk 3]
```diff
@@ -271,3 +282,56 @@ export function blockTagParamIndex(payload) {
...
+});
+
+export const stripHexPrefix = (str) => {
+  if (typeof str !== 'string') {
+    return str;
+  }
+  return isHexPrefixed(str) ? str.slice(2) : str;
+};
+
+const buildTransactionCommon = (txMeta) =>
```
Comment (gantunesr) [inline app/util/networks/index.js:298]: Can you add some unit tests around this new methods?

- `buildTxParams`
- `buildTransactionCommon`
- `buildUnserializedTransaction`
- `buildOVMGasPriceOracleContract`
- `fetchEstimatedL1Fee`

File: app/util/networks/index.js [hunk 4]
```diff
@@ -271,3 +282,56 @@ export function blockTagParamIndex(payload) {
...
+  ...omit(txMeta.txParams, 'gas'),
+  gasLimit: txMeta.txParams.gas,
+});
+
+export const stripHexPrefix = (str) => {
+  if (typeof str !== 'string') {
+    return str;
+  }
+  return isHexPrefixed(str) ? str.slice(2) : str;
+};
```
Comment (gantunesr) [inline app/util/networks/index.js:296]: This doesn't seem to be a network specific util method, probably more appropriate to have it in `conversions`, `address` or `number` files.

File: app/util/networks/index.js (hunk 3 above)
Comment (gantunesr) [reply app/util/networks/index.js:298]: Also, some of this methods seems to be more transactions than network related. You could probably create a new TypeScript file (for example, `optimismTransaction.ts`) inside `util/transactions`.

**Review by gantunesr — COMMENTED** [review]
Hey @dan437, I took a look at the PR and left some suggestions. Let me know what you think

File: ios/Podfile.lock (hunk 2 above)
Comment (dan437) [reply ios/Podfile.lock:857]: Without the new value my iOS build fails, but I can only keep the change locally and don't commit it.

File: app/components/UI/Swaps/QuotesView.js [hunk 5]
```diff
@@ -1565,6 +1586,26 @@ function SwapsQuotesView({
     handleQuotesErrorMetric(error);
   }, [error, handleQuotesErrorMetric, trackedError]);
 
+  useEffect(() => {
+    if (!multiLayerFeeNetwork || !selectedQuote?.trade) {
+      return;
+    }
+    const getEstimatedL1Fee = async () => {
+      try {
+        const eth = new Eth(Engine.context.NetworkController.provider);
```
Comment (Cal-L) [inline app/components/UI/Swaps/QuotesView.js:1595]: Probably don't need to include new `ethjs` library. You can use `ethjs-query` & `ethjs-contract`, which is already included in the app.
image

Comment (Cal-L) [reply app/components/UI/Swaps/QuotesView.js:1595]: Looks like `ethjs` depends on those libraries too so it's most likely doing the same thing
image

File: package.json [hunk 6]
```diff
@@ -181,6 +184,7 @@
     "ethereumjs-abi": "0.6.6",
     "ethereumjs-util": "6.1.0",
     "ethers": "^5.0.14",
+    "ethjs": "^0.4.0",
```
Comment (Cal-L) [inline package.json:187]: Can remove

**Review by Cal-L — COMMENTED** [review]
Left some comments

**Review by gantunesr — COMMENTED** [review]
LGTM. There are some unresolved comments by @Cal-L

**Comment by Cal-L** [comment]
Daniel, whenever you have the chance - Do you mind resolving the socket security check in the PR? You can just follow instructions from the Bot Commands to resolve

**Comment by dan437** [comment]
@SocketSecurity ignore node-hid@1.3.0
@SocketSecurity ignore node-hid@2.1.1
@SocketSecurity ignore usb@1.9.2

**Comment by dan437** [comment]
@SocketSecurity ignore node-hid@1.3.0 node-hid@2.1.1 usb@1.9.2

**Comment by dan437** [comment]
@SocketSecurity ignore node-hid@1.3.0

**Comment by gantunesr** [comment]
@SocketSecurity ignore node-hid@1.3.0
@SocketSecurity ignore node-hid@2.1.1
@SocketSecurity ignore usb@1.9.2

**Review by Cal-L — APPROVED** [review]
LGTM!

**Pull request merged**